distributed/dashboard/tests/test_scheduler_bokeh_html.py

```diff
@@ -119,10 +120,50 @@ def test_prometheus(c, s, a, b):
     clean_kwargs={"threads": False},
     scheduler_kwargs={"services": {("dashboard", 0): BokehScheduler}},
 )
-def test_health(c, s, a, b):
+async def test_prometheus_collect_task_states(c, s, a, b):
+    pytest.importorskip("prometheus_client")
+    from prometheus_client.parser import text_string_to_metric_families
+
+    http_client = AsyncHTTPClient()
+
+    # submit a task which should show up in the prometheus scraping
+    fut1 = c.submit(lambda: 1)
+
+    families = {}
+
+    await asyncio.sleep(0)
+
+    # We may need to scrape multiple times to allow the prometheus
+
+    bokeh_scheduler = s.services["dashboard"]
+    response = await http_client.fetch(
+        f"http://{bokeh_scheduler.server.address}:{bokeh_scheduler.port}/metrics"
+    )
```

Comment:
Perhaps the future hasn't yet had a chance to reach the scheduler?

Maybe something like this:

```python
future = client.submit(slowinc, 1, delay=0.5)

while not s.tasks:
   await asyncio.sleep(0.01)
```

Or, if you can allow for the future to be finished by the time you call this, then maybe ...

```python
future = client.submit(inc, 1)
await future
```

Reply:
I'll give it a shot

Reply:
Unfortunately, this also didn't work. There are scheduler tasks before and after the fetch but somehow it isn't picked up :(